sd/StringParser/src/com/company/StringFilter.java

```diff
@@ -0,0 +1,33 @@
+package com.company;
+import java.util.*;
+
+public class StringFilter {
+
+    public static String filter(String inputText, HashSet<String> blackList)
+    {
+        boolean firstPass = true;
+        Scanner scanner = new Scanner(inputText);
+        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
Could use StringJoiner sj = new StringJoiner(" "), in this case you wouldn't have to check for first pass